src/PerfView/PerfViewData.cs

```diff
@@ -5744,6 +5764,27 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                                     }
                                     sumCumMetric += cumMetric;
                                     cumCount++;
+
+                                    string GetAllocationType()
+                                    {
+                                        if (!allocationTypeNames.TryGetValue(callerIndex, out var typeName))
+                                        {
+                                            var cs = data.CallStack();
+                                            while (cs?.CodeAddress.ModuleFile is TraceModuleFile module)
+                                            {
+                                                if (allocationSites.TryGetValue(module, out var heapAllocationSites) &&
```

Comment:
Here is where I would put the lazy initialization of the heapAllocationSites (if you can't find the module file)  then populated it (a present but null value means its was populated as empty).